mm2src/coins/eth/v2_activation.rs

```diff
@@ -177,12 +188,6 @@ impl EthCoin {
             })
             .collect();
```

Comment:
Doesn't this create a new copy of the transport instead of sharing it with the platform coin. ref.
https://github.com/KomodoPlatform/komodo-defi-framework/blob/aab065eb3ac91fd49d0b5b79813a561e8e6feeec/mm2src/coins/eth/v2_activation.rs#L216
Will sharing this with platform coin change anything in GUI auth?

Reply:
When applying `Arc` suggestion, I forgot this part actually. We hold different tickers inside the web3 instances. Therefore we can't use the same web3 instances across eth contexes (platform coins and their tokens). I will revert the Arc from `web3_instances`.

Reply:
I agree to revert this for now. I will open an issue for this refactor since we should only open one connection at a time for the platform coin and all it's tokens to reduce number of open connections (similar to how we are trying to do for electrums ref https://github.com/KomodoPlatform/komodo-defi-framework/issues/791, https://github.com/KomodoPlatform/komodo-defi-framework/pull/1966). The issue will probably require some changes in GUI auth too, and maybe in nonce handling since I believe some of the nonce problems we faced before (not all) were due to sending Eth transactions to a different node than the token transactions which required waiting for transaction propagation across the network.

Reply:
Opened issue https://github.com/KomodoPlatform/komodo-defi-framework/issues/2071

Reply:
> The issue will probably require some changes in GUI auth too

Not necessarily needed. It's more related with the way of `web3` handles transports. We need to have gui-auth context inside of the transports, and gui-auth context is different for per ticker.

Reply:
ooh, I thought the ticker is also needed for proof of funding but I just checked [proof_of_funding.rs](https://github.com/KomodoPlatform/komodo-defi-proxy/blob/a83a3f31e414ea0b36d62e76e23684f8bac94dfe/src/security/proof_of_funding.rs#L27) and I understand now that only eth balance is used.

Reply:
I will handle the issue in the next sprint.